alternator/expressions.hh

```diff
@@ -60,24 +60,30 @@ enum class calculate_value_caller {
     UpdateExpression, ConditionExpression, ConditionExpressionAlone
 };
 
-inline std::ostream& operator<<(std::ostream& out, calculate_value_caller caller) {
-    switch (caller) {
-        case calculate_value_caller::UpdateExpression:
-            out << "UpdateExpression";
-            break;
-        case calculate_value_caller::ConditionExpression:
-            out << "ConditionExpression";
-            break;
-        case calculate_value_caller::ConditionExpressionAlone:
-            out << "ConditionExpression";
-            break;
-        default:
-            out << "unknown type of expression";
-            break;
-    }
-    return out;
 }
 
+template <> struct fmt::formatter<alternator::calculate_value_caller> {
+    constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+    auto format(alternator::calculate_value_caller caller, fmt::format_context& ctx) const {
+        std::string_view name = "unknown type of expression";
+        switch (caller) {
+            using enum alternator::calculate_value_caller;
+            case UpdateExpression:
+                name = "UpdateExpression";
+                break;
+            case ConditionExpression:
+                name = "ConditionExpression";
+                break;
+            case ConditionExpressionAlone:
+                name = "ConditionExpression";
+                break;
+        }
+        return fmt::format_to(ctx.out(), "{}", name);
+    }
+};
```

Comment:
A thought (won't prevent me from merging this PR):

Wouldn't it be nice if we wrote just once a feature that could allow printing any enum (or specific enums so designated), without need to repeat all the value names?

https://github.com/Neargye/magic_enum/blob/master/include/magic_enum/magic_enum.hpp has an example how you can with C++17 (plus some non-standard extensions on modern compilers) and there's an experimental https://en.cppreference.com/w/cpp/keyword/reflexpr, which could have given us a way to find an enum value's string name. But no rush in doing that :-)

Reply:
yeah, i looked into magic_enum long time ago. and it's the first time i put my eyes on reflexpr, thanks for point this out. but i am not sure if we want to use them. as they could be very controversial in this project. as 

1. magic_enum would introduce yet another dependency to this project. and it uses `__PRETTY_FUNCTION__`, which is a yet-standarized compiler feature. 
2. reflexpr as TS is not implemented by libstdc++  shipped along with GCC-13 yet. this is a deal breaker.

Reply:
Right. Note that if I was to use magic_enum, I wouldn't add it as a dependency - I would get "inspiration" from it and just implement exactly what we need, which is just a small subset of what magic_enum contains.
But we don't need to rush - if we wait a while longer maybe this reflexpr thing will become standard :-)

Reply:
i see. to ~~reinvent~~ reimplement it in scylla is a better option, as long as we don't insist on avoid using compiler specific extensions. yeah, i knew that `__PRETTY_FUNCTION__` is supported by both GCC and Clang. i will give it a stab.